chatkit-core/src/test/kotlin/com/pusher/chatkit/users/UserApiTypeMapperTest.kt

```diff
@@ -0,0 +1,76 @@
+package com.pusher.chatkit.users
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNull
+import com.pusher.chatkit.users.api.UserApiType
+import com.pusher.chatkit.users.api.UserApiTypeMapper
+import com.pusher.chatkit.util.DateApiTypeMapper
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+class UserApiTypeMapperTest : Spek({
+    describe("given a complete UserApiType") {
+        val userApiType = UserApiType(
+            id = "alice",
+            createdAt = "2020-03-05T15:31:21Z",
+            name = "alice",
+            customData = mapOf("item" to "data"),
+            avatarUrl = "http://placekitten.com/200/200",
+            updatedAt = "2020-03-05T15:31:42Z"
+        )
+        describe("when mapped to a UserInternalType object") {
+            val dateApiTypeMapper = DateApiTypeMapper()
+            val userInternalType = UserApiTypeMapper(dateApiTypeMapper)
+                .toUserInternalType(userApiType)
+
+            it("then the id matches") {
+                assertThat(userInternalType.id).isEqualTo(userApiType.id)
+            }
+
+            it("then the name matches") {
+                assertThat(userInternalType.name).isEqualTo(userApiType.name)
+            }
+
+            it("then the custom data matches") {
+                assertThat(userInternalType.customData).isEqualTo(userApiType.customData)
+            }
+
+            it("then the avatar url matches") {
+                assertThat(userInternalType.avatarUrl).isEqualTo(userApiType.avatarUrl)
+            }
+
+            it("then the created at matches") {
+                assertThat(userInternalType.createdAt).isEqualTo(1583422281000L)
+            }
+
+            it("then the updated at matches") {
+                assertThat(userInternalType.updatedAt).isEqualTo(1583422302000L)
+            }
+        }
+    }
+
+    describe("given a partially complete UserApiType object") {
+        val userApiType = UserApiType(
+            id = "alice",
+            createdAt = "2020-03-05T15:31:21Z",
+            name = "alice",
+            customData = null,
+            avatarUrl = null,
+            updatedAt = "2020-03-05T15:31:42Z"
+        )
```

Comment:
```suggestion
        val userApiType = UserApiType(
            id = "alice",
            createdAt = "2020-03-05T15:31:21Z",
            updatedAt = "2020-03-05T15:31:42Z"
        )
```
consistent order, brevity. we shall be then also checking the absence of name within the tests below

Reply:
This code won't compile, we're not using defaults = null, so we need to pass them through. They're in the same order as the previous test is in. So I'm not sure what we can do differently here.

Reply:
oh sorry my CPU must had been busy and the IDE didn't show red quickly enough for me to realise! when it comes to the order though I meant the order matching the order in constructor (in particular having createdAt/updatedAt grouped).

Reply:
Ah. Okay, i've put the createdAt and updatedAt together.